src/App.tsx

```diff
@@ -1,27 +1,72 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [visiableGoods, setVisiableGoods] = useState<Good[]>([]);
+  const [showGoods, setShowGoods] = useState(false);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const handelLoadAllGoods = useCallback(
+    () => {
+      getAll()
+        .then(result => setVisiableGoods(result))
+        .catch(() => setShowGoods(false))
+        .finally(() => setShowGoods(true));
+    },
+    [],
+  );
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handelLoad5FirstGoods = useCallback(
+    () => {
+      get5First()
+        .then(result => setVisiableGoods(result))
+        .catch(() => setShowGoods(false))
+        .finally(() => setShowGoods(true));
```

Comment:
what does finally do here?
i think it will work incorrectly if there is an error.
showGoods will always be true, even if there is an error